lib/rewrite/rewrite-expr-grammar.ym

```diff
@@ -158,9 +159,14 @@ rewrite_subst_opt
 
 rewrite_expr_opts
         : rewrite_expr_opt rewrite_expr_opts
+        | rewrite_set_opt rewrite_expr_opts
         |
         ;
 
+rewrite_set_opt
+        : template_option
+        ;
+
 rewrite_expr_opt
```

Comment:
This way `template_option` is only enabled to `rewrite-set`. Probably I can rephrase this to a simpler form.